### PR TITLE
meson: do not fail if rsync is not installed with meson 0.57.2

### PR DESCRIPTION
**Pull request by keszybz**
https://github.com/mesonbuild/meson/issues/8641

Our CI started to fail. Even if the change is reverted in meson,
we need a quick workaround here.

### CODE REVIEW

File: man/meson.build [hunk 1]
```diff
@@ -184,17 +184,20 @@ html = custom_target(
...
-        depends : man_pages + html_pages,
-        command : ['rsync', '-rlv',
-                   '--delete-excluded',
-                   '--include=man',
-                   '--include=*.html',
-                   '--exclude=*',
-                   '--omit-dir-times',
-                   meson.current_build_dir(),
-                   get_option('www-target')])
+rsync = find_program('rsync')
```
Comment (bluca) [inline man/meson.build]: `required: false` ?

> Program rsync found: NO
>
> man/meson.build:187:0: ERROR: Program 'rsync' not found

Comment (keszybz) [reply man/meson.build]: Duh.

**Review by bluca — APPROVED** [review]
lgtm

File: man/meson.build [hunk 2]
```diff
@@ -184,17 +184,20 @@ html = custom_target(
...
-        depends : man_pages + html_pages,
-        command : ['rsync', '-rlv',
-                   '--delete-excluded',
-                   '--include=man',
-                   '--include=*.html',
-                   '--exclude=*',
-                   '--omit-dir-times',
-                   meson.current_build_dir(),
-                   get_option('www-target')])
+rsync = find_program('rsync', required : 'false')
```
Comment (bluca) [inline man/meson.build]: actually lgtm-not:

> man/meson.build:187:0: ERROR: required keyword argument must be boolean or a feature option

shouldn't be a string

Comment (keszybz) [reply man/meson.build]: Sorry.

Comment (bluca) [reply man/meson.build]: third time's the charm!

**Comment by bluca** [comment]
that's weird, pakit is still failing:

```
Program rsync found: NO

man/meson.build:187:0: ERROR: Program 'rsync' not found
```

**Comment by mrc0mmand** [comment]
The Packit fail should be fixed by #19299 - @keszybz could you rebase the PR against main just to be sure?

**Comment by keszybz** [comment]
Rebased.

**Comment by mrc0mmand** [comment]
Oh well, now we get to enjoy the `-Wmaybe-uninitialized` fallout, since the CI finally tests the correct code...

```
In file included from ../src/basic/efivars.c:13:
../src/basic/efivars.c: In function ‘cache_efi_options_variable’:
../src/basic/alloc-util.h:48:9: error: ‘k’ may be used uninitialized [-Werror=maybe-uninitialized]
   48 |         free(memory);
      |         ^~~~~~~~~~~~
../src/basic/efivars.c:353:38: note: ‘k’ was declared here
  353 |                 _cleanup_free_ char *k;
      |                                      ^
cc1: all warnings being treated as errors
```

**Comment by bluca** [comment]
That's a separate issue from `rsync`, so let's merge this

**Comment by bluca** [comment]
For the uninit warning: https://github.com/systemd/systemd/pull/19302

**Pull request merged**